azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -733,6 +773,23 @@ def _report_agent_cgroups_procs(agent_cgroup_proc_names, unexpected):
                     msg = "Agent includes following processes when UNKNOWN process found: {0}".format("\n".join([ustr(proc) for proc in agent_cgroup_proc_names]))
                     add_event(op=WALAEventOperation.CGroupsInfo, message=msg)
 
+        @staticmethod
+        def _report_agent_cgroups_memory_usage(cgroup_metrics):
+            """
+            Reports the agent's current memory usage when it exceeds the configured limit.
+            """
+            limit_in_bytes = conf.get_agent_memory_quota()
+            current_usage = 0
+            for metric in cgroup_metrics:
+                if metric.counter == MetricsCounter.TOTAL_MEM_USAGE:
```

Comment:
could we combine this if/elif into one if condition?

```suggestion
                if metric.counter == MetricsCounter.TOTAL_MEM_USAGE or metric.counter == MetricsCounter.SWAP_MEM_USAGE:
```